app/assets/javascripts/discourse/app/models/topic.js

```diff
@@ -851,7 +853,23 @@ Topic.reopenClass({
       data.topic_ids = topicIds;
     }
 
-    return ajax("/topics/reset-new", { type: "PUT", data });
+    const user = User.current();
+    if (user.new_new_view_enabled) {
+      data.dismissTopics = true;
+      data.dismissPosts = true;
+      showModal("dismiss-new", {
+        model: data,
+        titleTranslated: I18n.t("topics.bulk.dismiss_new_modal.title"),
+      });
+    } else {
+      return ajax("/topics/reset-new", { type: "PUT", data }).then(() => {
+        const controller = getOwner(this).lookup("controller:discovery/topics");
+        controller.send(
+          "refresh",
+          tracked ? { skipResettingParams: ["filter", "f"] } : {}
+        );
+      });
+    }
```

Comment:
I think making the ajax call be should be a task for the `Topic` model exclusively (there's a duplicate call the modal controller `controllers/dismiss-new.js` above) and the model shouldn't be responsible for showing the modal.

Can we refactor this so:

1. there's only 1 place where the ajax call is performed (the model), and
2. the `showModal` function is called outside/before `Topic.resetNew` is called

? We may need a new mixin to contain the modal displaying logic because there're 2 places where the modal can be triggered (`controllers/tag-show.js` and `controllers/discovery/topics.js`)